restapi/client.go

```diff
@@ -418,6 +418,16 @@ func newS3BucketClient(claims *models.Principal, bucketName string, prefix strin
 	return s3Client, nil
 }
 
+// pathJoinFinalSlash - like path.Join() but retains trailing slashSeparator of the last element
+func pathJoinFinalSlash(elem ...string) string {
+	if len(elem) > 0 {
+		if strings.HasSuffix(elem[len(elem)-1], "/") {
+			return path.Join(elem...) + "/"
```

Comment:
```suggestion
			return path.Join(elem...) + slashSeparator
```

Prefer more constants as I mentioned before